content/en/docs/instrumentation/js/resources.md

```diff
@@ -0,0 +1,247 @@
+---
+title: "Resources"
+weight: 6
+description:
+---
+
+A [resource][] is an immutable representation of the entity producing telemetry
+as attributes. For example, a process producing telemetry that is running in a
+container on Kubernetes has a Pod name, it is in a namespace and possibly is
+part of a deployment which also has a name. All three of these attributes can be
+included in the resource.
+
+In your observability backend you can use the information stored in the resource
+to see to which container (or pod, or deployment, etc.) your traces belong and
+if they have issues (like out of memory, network outages, etc.) you can quickly
+understand why your traces might be slower or errounous.
+
+Below you will find some introductions on how to setup resource detection with
+the Node.JS SDK
+
+## Setup
+
+Follow the instructions in the [Getting Started - Node.js][], so that you have
+the files `package.json`, `app.js` and `tracing.js`.
+
+## Process & Environment Resource Detection
+
+Out of the box the Node.JS SDK detects the [process and process runtime
+resources][] and takes attributes from the environment variable
+`OTEL_RESOURCE_ATTRIBUTES`. A quick way to verify that, turn on debugging in
+`tracing.js`:
+
+```javascript
+// For troubleshooting, set the log level to DiagLogLevel.DEBUG
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG);
+```
+
+Run the application with some values set to `OTEL_RESOURCE_ATTRIBUTES`, e.g. we
+set the `host.name` to identify the [Host][]:
+
+```sh
+$ env OTEL_RESOURCE_ATTRIBUTES="host.name=localhost" node --require './tracing.js' app.js
+@opentelemetry/api: Registered a global for diag v1.2.0.
+...
+Listening for requests on http://localhost:8080
+EnvDetector found resource. Resource { attributes: { 'host.name': 'localhost' } }
+ProcessDetector found resource. Resource {
+  attributes: {
+    'process.pid': 12345,
+    'process.executable.name': 'node',
+    'process.command': '/app.js',
+    'process.command_line': '/bin/node /app.js',
+    'process.runtime.version': '16.17.0',
+    'process.runtime.name': 'nodejs',
+    'process.runtime.description': 'Node.js'
+  }
+}
+...
+```
+
+As you can see the SDK has detected the process and also added the attributes
+you set via the environment variable automatically.
+
```

Comment:
Suggestion: add a nested subsection that makes this about adding your own resources, like:

```markdown
### Adding your own resources with environment variables
```

or

```markdown
### Adding your own resources with environment variables
```

And then that would have another section show how to do it in code instead.

Reply:
Great point, I missed that, updated !